spec/models/reservations_validations_spec.rb

```diff
@@ -60,4 +62,20 @@
       end
     end
   end
+
+  describe "conditionally run validations" do
+    context "with reserve_start_at, reserve_end_at, and reservation_changed? is true" do
```

Comment:
This is an interesting question. I sort of feel like we can depend on `setup_reservation` not changing, because so many tests depend on it. Whether or not that is desirable in our test structure seems debatable, but given that we're pretty firmly placed in that paradigm, I feel like we might as well take advantage of the abstraction.